### DISCUSSION TITLE
Main window launching options

### DISCUSSION THREAD

**Discussion by majkinetor** [1Remote/1Remote#417] [Ideas]
Currently we have ALT+m to run launcher. I think it makes sence to allow using full 1RM UI as a sort of launcher with few quick modifications.

1. Introduce a new global shortcut to show main window (part of #405)
2. ~~Automatically focus search field on showing (currently requires CTRL + F)~~ [already works this way]
3. Allow using up/down arrow to go over results (highlight), while typing, without loosing search focus. We need to have this anyway, as 1RM is not keyboard friendly then and can't be used in accessible manner fully.
4. Allow using <ENTER> in search to launch the selected connection (currently does nothing)
    5. If no connection is selected, use the first one shown
    6. We also need to have this anyway due to accessibility
5. Automatically minimize 1RM upon selection to tray 

This is all very easy to implement, probably in an hour or so and 3 of those are something we need to do anyway (1,3,4)

Everything can be done without introducing new options except the last one, as it may not be what some people want. So we would need either global option for this, or better, something more handy so you can quickly toggle it in different contexts. Which leads me to ad hoc launcher options in the main window:

![image](https://github.com/1Remote/1Remote/assets/85767/7ddc7fc3-f91d-4148-a3bd-865c2ae7ce20)

On launch:

1. Minimize 
2. Keep focused

Explanation:

1. First option would minimize 1RM to tray upon launching connection with ENTER. It will then behave like the ALT + M Launcher. I hate it when i run connections from the main window (and I do that quite often) and then it stays open, so now clicking with the mouse the 1RM taskbar icon shows 2 windows so I need to manually close 1RM main window in order for it not do so. This will both eliminate this unfortunate behavior and allow us to use it as a new type of launcher. We already know that some people don't like to use ALT M launcher <details><summary>details</summary> ![image](https://github.com/1Remote/1Remote/assets/85767/3f183acb-1d48-4efc-b42f-143e23f60843) </details>
2. Second option could be used when you want to open several connections at the same time. Currently, when you open 1 connection it is focued so in order to run more of them you need to manually return to the main window, which is cumbersome, especially if one of them is in full screen. This option would allow you to open connections in the background. Note that this option doesn't make any sense in combination with the first one.

Those are the basics, but could be improved further with the following:

1. Show running connections in the main window (we could use #416)
2. Use a hotkey modifier CTRL (ENTER or mouse) to decide if launching will be in the background or not

**Comment by VShawn** [comment]
> Automatically focus search field on showing (currently requires CTRL + F)

FYI you dont actually press ctrl+f, just type the keyword.

> Minimize, Keep focused

Can we place this two options in `settings` named `Main windows behave`? and better names are needed for the two options to make their functions can be better understood.

**Comment by majkinetor** [comment]
You dont wont 2nd one all the time. First one u may want all the time.

Lets go this way:

1. Ist one in global options. 
2. No option for sexond one but modify with  CTRL click/enter. It also prevents option 1

**Comment by majkinetor** [comment]
> FYI you dont actually press ctrl+f, just type the keyword.

Didn't notice that, even better, then step 2 is already done.

> Can we place this two options in settings named Main windows behave?

I put them there because it was really something you might wanna change more frequently, in ad hoc manner. With [above](https://github.com/1Remote/1Remote/discussions/417#discussioncomment-5901887) option it would work like this:

- [x] After launching a connection, minimize the main window to tray

App will go to tray when you dblclick connection or use enter. If you hold CTRL it will not minimize and connection will be launched in the background.


- [ ] After launching a connection, minimize the main window to tray

App will not go to tray when you dblclick connection or use enter. If you hold CTRL connection will be launched in the background. There is no option to minimize on launch, you need to switch it in the settings.

**Reply by VShawn** [reply]
> Allow using up/down arrow to go over results

Do not forget there are fans of card view, how it works with `card view`?

**Reply by majkinetor** [reply]
it goes left to right first :) E.g. it means next result, not up/down. In card view you could actually have up/down/left/right... but who cares :)

**Comment by majkinetor** [comment]
Since I am not sure if you are OK with discussions, how about you 👍  the first post if you wanna go for it, or 👎 if not. Then I can create appropriate ticket with the stuff we agreed both ? 

1. If we have 50%+ we close the discussion and create implementation ticket.
2. If we are 50% or lower we leave discussion open. 
3. If there are 0% we close discussion as wont-fix
